### ISSUE TITLE
Malformed SVG: missing `d` attribute on a `path`

### ISSUE THREAD

**Issue by rachel-fenichel** [LLK/scratch-blocks#1217]
Bug found on pxt-blockly, which uses scratch-blocks rendering: https://github.com/Microsoft/pxt-blockly/pull/86 and https://github.com/Microsoft/pxt-blockly/issues/84

### Expected Behavior

Blocks load in IE.

### Actual Behavior

Some blocks can't load in IE 11.  In Chrome there is a `path` with no `d` attribute, but Chrome recovers cleanly.
![image](https://user-images.githubusercontent.com/13686399/32580973-d540a6c6-c49c-11e7-9e48-6199b9a2543c.png)

The problem occurs when loading from XML.  A block with a connected input with no shadow block will fail; a block with a connected input with a shadow block won't fail.  The problem has to do with the rendering of the `inputShape` on the input with no shadow block.

### Steps to Reproduce

In the vertical playground, import this code:
```
<xml xmlns="http://www.w3.org/1999/xhtml">
  <variables></variables>
  <block type="operator_and" id="{o/zPw8mbvA4/K#l:DY#" x="219" y="340">
    <value name="OPERAND2">
      <block type="operator_not" id=";;WvgQM]fo%[WIC=JHj$"></block>
    </value>
  </block>
  <block type="operator_multiply" id="qIedr}Oh7)bU:M)_1i?~" x="239" y="431">
    <value name="NUM1">
      <shadow type="math_number" id="5@*vu7.;lmFcBi7Skbnr">
        <field name="NUM"></field>
      </shadow>
    </value>
    <value name="NUM2">
      <shadow type="math_number" id=")TRj+SV/SRUKtZf,DWIO">
        <field name="NUM"></field>
      </shadow>
      <block type="operator_round" id="3k$HdNFyy9(QAA_ZfW*L">
        <value name="NUM">
          <shadow type="math_number" id="l;VaN.ZqP_bJzyh,?T4v">
            <field name="NUM"></field>
          </shadow>
        </value>
      </block>
    </value>
  </block>
</xml>
```

It should look like this: 
![image](https://user-images.githubusercontent.com/13686399/32580922-8905cc0a-c49c-11e7-9139-67604f3ae563.png)

Without dragging any blocks, open up developer tools->elements.

Inspect the paths on the blocks above.

### Operating System and Browser

Visible problem on IE 11 when loading from XML, but the problem is present on Chrome and other browsers--it just recovers nicely.

**Comment by samelhusseini** [comment]
Just to clarify, it's not load that is blocked on IE. but SVG calculations (for the scrollbar height, getBBox) and applied filters (disappearing as you drag) that are affected.

**Comment by rachel-fenichel** [comment]
Ah good point.  But creating the `and` block above by dragging won't cause the same problem--the input-without-a-shadow has to be filled at the point of the first render.  If the block was rendered at least once with the input unfilled, it will set the path: https://github.com/LLK/scratch-blocks/blob/develop/core/block_render_svg_vertical.js#L1335

Covering it later sets the path to `visibility: hidden`, but doesn't remove the `d` attribute: https://github.com/LLK/scratch-blocks/blob/develop/core/block_render_svg_vertical.js#L1324

**Comment by rachel-fenichel** [comment]
Aha, and here's the code that just avoids creating input shapes if there's a shadow.
```
Blockly.BlockSvg.prototype.initInputShape = function(input) {
  if (this.inputShapes_[input.name] || input.connection.getShadowDom()) {
    // Only create the shape elements once, and don't bother creating them if
    // there's a shadow block that will always cover the input shape.
    return;
  }
  this.inputShapes_[input.name] = Blockly.utils.createSvgElement(
    'path',
    {
      'class': 'blocklyPath',
      'style': 'visibility: hidden' // Hide by default - shown when not connected.
    },
    this.svgGroup_
  );
};
```

**Comment by rachel-fenichel** [comment]
Fixed by #1219
